ecommerce/frontend/src/components/VendorContent/VendorSplash.js

```diff
@@ -0,0 +1,61 @@
+        description: 'Create the future, Save the planet ',
+        rating: 8.7,
+    }
+    const { title, imageUrl, description, rating } = vendorHeaderDetails
+    const { user } = useAppContext()
+    const isVendorAndOwner = user.type === 'vendor' && vendorId === user.id.toString()
+    return (
+        <div className="vendor-splash">
+            <div className="vendor-image">
+                <img src={imageUrl} className="splash-logo"></img>
+            </div>
+            <div className="vendor-header">
+                <div className="vendor-header-content">
+                    <h1 className="vendor-name">{title}</h1>
+                    <h3 className="vendor-slogan">{description}</h3>
+                    <div
+                        className={`product-header-vendor-rating product-header-vendor-rating__${getVendorRatingLevel({
+                            rating,
+                        })}`}>
+                        {round(rating)}
+                    </div>
+                </div>
+                <div className="vendor-edit-button">
+                    {isVendorAndOwner ? (
+                        <div className="vendor-edit-mode-open">
+                            <p>{editable ? 'Edit Mode' : 'Seeing as Customer'}</p>
+                            <Switch defaultChecked={true} onChange={onEditModeChange} />
+                        </div>
+                    ) : (
+                        <Button type="primary" icon={<EditOutlined />} href="/profile/messages">
+                            Send a message
+                        </Button>
```

Comment:
Not sure if Button has href but in any case a regular html link shouldn't be used. You need to use either Link from react-router or have a onClick which uses `history.push('/profile/messages/`). 

Also messages page does not allow creating a new conversations. For this you will need to use a  "new message modal". We are working on it with @ozdenizdolu in F-prepare-vendor-ui branch.

Reply:
Ok, then I will use onClick function, which first opens a modal for 'Sending a Message', after sending a message function uses 
`history.push('/profile/messages/').` and finishes.